demos/plot_different_optimization_methods.py

```diff
@@ -28,6 +28,7 @@
     times = [sum(t[:i]) for i in range(len(t))]
     axes.loglog(times, J, label=method)
 axes.grid(True)
+axes.set_title(f"{demo} with mesh adaptation by opt_{run}.py")
```

Comment:
I don't like this title. It would be better to avoid references to file names in plots. You could do something like
```
labels = {"uniform": "a uniform mesh", "hessian": "Hessian-based mesh adaptation", "go": "goal-oriented mesh adaptation"}
title = f"{demo} solved using {labels[run]}"
```